core/src/commonMain/kotlin/com/sourcepoint/mobile_core/storage/Repository.kt

```diff
@@ -47,17 +31,17 @@ class Repository(private val storage: Settings) {
             .associateWith { storage[it]!! }
         set(value) {
             storage.removeKeysStartingWith(prefix = GPP_PREFIX)
-            value.entries.forEach {
-                storage.setJsonPrimitive(it.key, it.value)
-            }
+            value.entries.forEach { storage[it.key] = it.value }
         }
 
     var uspString: String?
         get() = storage[USPSTRING_KEY]
         set(value) { storage[USPSTRING_KEY] = value }
 
     var state: State?
-        get() = Json.decodeFromString<State?>(storage.getString(SP_STATE_KEY, defaultValue = ""))
+        get() = runCatching {
```

Comment:
@andresilveirah can you please add `@Throws(Exception::class)` so iOS could expect and catch an exception properly?

Reply:
@Nevazhnovu but this code shouldn't throw any exception. The `runCatching` will catch the xception and the `getOrNull` will return `null` if an exception is thrown.